### PR TITLE
Add a contravariant functor instance to ObjectEncoder

### PR DESCRIPTION
**Pull request by backuitist**

### CODE REVIEW

**Comment by travisbrown** [comment]
Thanks @backuitist!

File: modules/core/shared/src/main/scala/io/circe/ObjectEncoder.scala [hunk 1]
```diff
@@ -41,6 +50,13 @@ final object ObjectEncoder extends LowPriorityObjectEncoders {
   final def instance[A](f: A => JsonObject): ObjectEncoder[A] = new ObjectEncoder[A] {
     final def encodeObject(a: A): JsonObject = f(a)
   }
+
+  /**
+    * @group Instances
+    */
+  implicit final val encoderContravariant: Contravariant[ObjectEncoder] = new Contravariant[ObjectEncoder] {
```
Comment (backuitist) [inline modules/core/shared/src/main/scala/io/circe/ObjectEncoder.scala]: Done

**Comment by backuitist** [comment]
Thanks for the quick review @travisbrown

**Comment by travisbrown** [comment]
Thanks, @backuitist! The test failure looks like an unrelated fluke (the string `꩘` apparently parses as a valid Java integer)—I'll fix it in a separate PR, restart the test for now, and merge when green.

**Pull request merged**